ArcGISRuntimeSDKQt_CppSamples/Search/OfflineGeocode/OfflineGeocode.cpp

```diff
@@ -314,19 +314,22 @@ void OfflineGeocode::onGeocodingCompleted_(const QList<GeocodeResult>& geocodeRe
   m_geocodeInProgress = false;
   emit geocodeInProgressChanged();
 
-  // if there are no matching results, notify user
-  if (geocodeResults.empty())
+  // dismiss callouts
+  m_calloutData->setVisible(false);
+  m_pinGraphic->setVisible(false);
+
+  // if there are no matching results, notify user and stop processing
+  if (geocodeResults.isEmpty())
   {
     m_noResults = true;
     emit noResultsChanged();
+    return;
```

Comment:
This is the real fix. Obviously we should call `geocodeResults.at(0)` after we've determined the QList is empty.